pkgs/build-support/mkshell/minimal.nix

```
@@ -0,0 +1,47 @@
+{ writeTextFile, writeScript, bash }:
+
+# A special kind of derivation that is only meant to be consumed by the
+# nix-shell. This differs from the traditional `mkShell` in that it is
+# ultra minimalist in dependencies.
+# It does not come with traditional stdenv (i.e. coreutils).
+# Its only dependency is essentially bash.
+{
+# a list of packages to add to the shell environment
+# we simplify here rather than `native` or `propagated` since
+# this is only ever used to be in a nix-shell; which is by default
+# the current system
+packages ? [ ], ... }@attrs:
+
+derivation ({
+  name = "minimal-nix-shell";
+
+  # What is the need for this reference to stdenv?
+  # Well Nix itself, checks to see if this attribute exists
+  # and sources the file.
+  # We will setup our nix-shell environment here and do things
+  # like clear the $PATH
+  # reference: https://github.com/NixOS/nix/blob/94ec9e47030c2a7280503d338f0dca7ad92811f5/src/nix-build/nix-build.cc#L494
+  "stdenv" = writeTextFile rec {
+    name = "setup";
+    executable = true;
+    destination = "/${name}";
+    text = ''
+      set -e
+      unset PATH
+
+      for a in $packages; do
+        export PATH=$a/bin:$PATH
+      done
+    '';
+  };
+
+  builder = writeScript "builder.sh" ''
+    #!${bash}/bin/bash
+    echo
+    echo "This derivation is not meant to be built, aborting";
+    echo
+    exit 1
+  '';
+
+  system = builtins.currentSystem;
+} // attrs)
```

So all attrs behave like `passthru`? That diverges a lot from normal nix-shell, which I don't think is desirable.
May I suggest the following for tooling to pick up on as well:

```suggestion
} // {
  isShell = true;
  phases = ["nobuildPhase"];
} // (attrs.passthru or {}))
```

---

Although using a default argument is cleaner imo.

---

well you need to be able to set environment variables so those should just be merged into the derivation.
Not sure if only merging _passthru_ is then correct.

---

Oops, I misread and held on to my mistake in the suggestion.